src/syntax_macros.rs

```diff
@@ -0,0 +1,223 @@
+#[macro_export]
+macro_rules! num {
+    ($f:ty, $i:literal) => {
+        $crate::syntax::Syntax::<$f>::Num(Pos::No, ($i as u64).into())
+    };
+    ($i:literal) => {
+        $crate::syntax::Syntax::Num(Pos::No, ($i as u64).into())
+    };
+    ($i:expr) => {
+        $crate::syntax::Syntax::Num(Pos::No, $i)
+    };
+}
+
+#[macro_export]
+macro_rules! uint {
+    ($f:ty, $i:literal) => {
+        $crate::syntax::Syntax::<$f>::UInt(Pos::No, $crate::uint::UInt::U64($i as u64))
+    };
+    ($i:literal) => {
+        $crate::syntax::Syntax::UInt(Pos::No, $crate::uint::UInt::U64($i as u64))
+    };
+}
+
+#[macro_export]
+macro_rules! str {
+    ($f:ty, $i:literal) => {
+        $crate::syntax::Syntax::<$f>::String(Pos::No, $i.to_string())
+    };
+    ($i:literal) => {
+        $crate::syntax::Syntax::String(Pos::No, $i.to_string())
+    };
+}
+
+#[macro_export]
+macro_rules! char {
+    ($f:ty, $i:literal) => {
+        $crate::syntax::Syntax::<$f>::Char(Pos::No, $i as char)
+    };
+    ($i:literal) => {
+        $crate::syntax::Syntax::Char(Pos::No, $i as char)
+    };
+}
+
+#[allow(unused_macros)]
+#[macro_export]
+macro_rules! symbol {
+    ([$( $x:expr ),*]) => {
+        {
+            #[allow(unused_mut)]
+            let mut temp_vec = Vec::new();
+            $(
+                temp_vec.push($x.to_string());
+            )*
+            $crate::syntax::Syntax::Symbol(Pos::No, $crate::symbol::Symbol::Sym(temp_vec))
+        }
+    };
+    ($f:ty, [$( $x:expr ),*] ) => {
+        {
+            #[allow(unused_mut)]
+            let mut temp_vec = Vec::new();
+            $(
+                temp_vec.push($x.to_string());
+            )*
+            $crate::syntax::Syntax::<$f>::Symbol(Pos::No, $crate::symbol::Symbol::Sym(temp_vec))
+        }
+    };
+}
+
+#[allow(unused_macros)]
+#[macro_export]
+macro_rules! lurksymbol {
```

Comment:
Is there some example usage of these macros in general? Unless we *badly* need this `lurksymbol` macro and similar, I would like not to include it — because there should not be a syntax-level concept of a 'Lurk symbol'. I would rather we move forward and get to a correct solution embracing packages. My concern is that we use this fake concept of a 'lurk symbol' as a crutch which then *blocks* us solving the more general problem correctly.